storage/storage-mq/src/util/postgresRepository.ts

```diff
@@ -0,0 +1,75 @@
+import { Pool, PoolConfig, PoolClient, QueryResult } from 'pg'
+
+export default class PostgresRepository {
+  private connectionPool?: Pool = undefined
+
+  /**
+   * Initializes the connection to the database.
+   * @param retries:  Number of retries to connect to the database
+   * @param backoffMs:  Time in ms to backoff before next connection retry
+   * @returns reject promise on failure to connect
+   */
+  public async init (poolConfig: PoolConfig, retries: number, backoffMs: number): Promise<void> {
+    console.debug(`Connecting to database with config:\n${JSON.stringify(poolConfig)}`)
+
+    // try to establish connection
```

Comment:
I would extract the logic to connect to the DB with retries into a separate method.
Example:

```ts
public async init (poolConfig: PoolConfig, retries: number, backoffMs: number): Promise<void> {
    console.debug('Initializing databse connection')
    console.debug(`Connecting to database with config:\n${JSON.stringify(poolConfig)}`)

    this.connectionPool = connectWithRetries(poolConfig, MAX_RETRIES)

    if (!this.connectionPool) {
      return Promise.reject(new Error('Connection to databse could not be established.'))
    }

    console.info('Successfully established connection to database.')
    return Promise.resolve()
}

private connectWithRetries (poolConfig, retries): connectionPool{
 for (let i = 1; i <= retries; i++) {
    console.info(`Initializing database connection (${i}/${retries})`)
    let client
    try {
      const connectionPool = new Pool(poolConfig)
      client = await connectionPool.connect()
      await client.query('SELECT 1')
      console.info('Successfully established database connection')
      return connectionPool
    } catch (error) {
      await this.sleep(backOffMs)
    } finally {
      if (client) {
        client.release()
      }
    }
  }
}
```